internal/sql/repository/chartConfig/ChartRepository.go

```diff
@@ -73,6 +118,151 @@ type ChartRepositoryImpl struct {
 	dbConnection *pg.DB
 }
 
+func (repositoryImpl ChartRepositoryImpl) FindBatchOperationExample(task string) (*BatchOperationExample, error) {
+	batchOperationExample := &BatchOperationExample{}
+	err := repositoryImpl.dbConnection.
+		Model(batchOperationExample).Where("task LIKE ?", task).
+		Select()
+	return batchOperationExample, err
+}
+
+func (repositoryImpl ChartRepositoryImpl) FindBulkAppNameIsGlobal(appNameIncludes []string, appNameExcludes []string) ([]*App, error) {
+	var apps []*App
+	//Concatenating string according to sql LIKE operator required format
+	var appNameIncludesQuery string
+	for i, appNameInclude := range appNameIncludes {
+		if i == 0 {
+			appNameIncludesQuery += fmt.Sprintf("app_name LIKE '%s' ", appNameInclude)
+		} else {
+			appNameIncludesQuery += fmt.Sprintf("OR app_name LIKE '%s' ", appNameInclude)
+		}
+	}
+	var appNameExcludesQuery string
+	for i, appNameExclude := range appNameExcludes {
+		if i == 0 {
+			appNameExcludesQuery += fmt.Sprintf("app_name NOT LIKE '%s' ", appNameExclude)
+		} else {
+			appNameExcludesQuery += fmt.Sprintf("AND app_name NOT LIKE '%s' ", appNameExclude)
+		}
+	}
+	appNameQuery := fmt.Sprintf("%s AND %s", appNameIncludesQuery, appNameExcludesQuery)
```

Comment:
what if there len(appNameExcludesQuery) == 0, this will be invalid query

Reply:
added check for this